src/gluonts/mx/trainer/callback.py

```diff
@@ -0,0 +1,553 @@
+        trainer
+            The trainer which is running the training.
+
+        Returns
+        -------
+        bool
+            A boolean whether the training should continue. Defaults to `True`.
+        """
+        return True
+
+    def on_epoch_end(
+        self,
+        epoch_no: int,
+        epoch_loss: float,
+        training_network: nn.HybridBlock,
+        trainer: gluon.Trainer,
+        best_epoch_info: Dict[str, Any],
+        ctx: mx.Context,
+    ) -> bool:
+        """
+        Hook that is called after every epoch. As `on_train_epoch_end` and
+        `on_validation_epoch_end`, it returns a boolean whether training should continue. This
+        hook is always called after `on_train_epoch_end` and `on_validation_epoch_end`. It is
+        called regardless of these hooks' return values.
+
+        Parameters
+        ----------
+        epoch_no
+            The current epoch (the first epoch has `epoch_no = 0`).
+        epoch_loss
+            The validation loss that was recorded in the last epoch if validation data was
+            provided. The training loss otherwise.
+        training_network
+            The network that is being trained.
+        trainer
+            The trainer which is running the training.
+        best_epoch_info
+            Aggregate information about the best epoch. Contains keys `params_path`, `epoch_no` and
+            `score`. The score is the best validation loss if validation data is provided or the
+            best training loss otherwise.
+        ctx
+            The MXNet context used.
+
+        Returns
+        -------
+        bool
+            A boolean whether the training should continue. Defaults to `True`.
+        """
+        return True
+
+    def on_train_end(
+        self,
+        training_network: nn.HybridBlock,
+        temporary_dir: str,
+        ctx: mx.context.Context = None,
+    ) -> None:
+        """
+        Hook that is called after training is finished. This is the last hook to be called.
+
+        Parameters
+        ----------
+        training_network
+            The network that was trained.
+        temporary_dir
+            The directory where model parameters are logged throughout training.
+        ctx
+            An MXNet context used.
+        """
+
+
+class CallbackList(Callback):
+    """
+    Used to chain a list of callbacks to one Callback.
+    Boolean hook methods are logically joined with AND, meaning that if at least one callback method returns False, the training is stopped.
+
+    Parameters
+    ----------
+    callbacks
+        A list of gluonts.mx.trainer.callback.Callback's.
+    """
+
+    @validated()
+    def __init__(self, callbacks: Union[List[Callback], Callback], **kwargs):
+
+        self.callbacks = (
+            callbacks if isinstance(callbacks, list) else [callbacks]
+        )
+
+    def include(
+        self, new_callbacks: Union["CallbackList", List[Callback]]
```

Comment:
I'm really not a fan of this. The interface should be clear and simple, so mixing different types like that is something I would avoid.